packages/aws-rfdk/lib/core/scripts/bash/mountEfs.sh

```diff
@@ -49,31 +55,104 @@ else
 fi
 
 function use_amazon_efs_mount() {
-  test -f "/sbin/mount.efs" || sudo "${PACKAGE_MANAGER}" install -y "${AMAZON_EFS_PACKAGE}"
+  test -f "/sbin/mount.efs" || "${PACKAGE_MANAGER}" install -y "${AMAZON_EFS_PACKAGE}"
   return $?
 }
 
 function use_nfs_mount() {
-  test -f "/sbin/mount.nfs4" || sudo "${PACKAGE_MANAGER}" install -y "${NFS_UTILS_PACKAGE}"
+  test -f "/sbin/mount.nfs4" || "${PACKAGE_MANAGER}" install -y "${NFS_UTILS_PACKAGE}"
   return $?
 }
 
+function resolve_mount_target_ip_via_api() {
+  local EFS_FS_ID=$1
+  local MNT_TARGET_RESOURCE_ID=$2
+  local AVAILABILITY_ZONE_NAME=$3
+  local AWS_REGION=$4
+  local MOUNT_POINT_IP=""
+
+  if [[ $MNT_TARGET_RESOURCE_ID == fs-* ]]
+  then
+    # Mounting without an access point
+    MOUNT_POINT_IP=$(aws efs describe-mount-targets \
+      --region "${AWS_REGION}"                      \
+      --file-system-id ${MNT_TARGET_RESOURCE_ID}    \
+      | jq -r ".MountTargets[] | select( .AvailabilityZoneName == \"${AVAILABILITY_ZONE_NAME}\" ) | .IpAddress" \
+    )
+  elif [[ $MNT_TARGET_RESOURCE_ID == fsap-* ]]
+  then
+    # Mounting via an access point
+    MOUNT_POINT_IP=$(aws efs describe-mount-targets \
+      --region "${AWS_REGION}"                      \
+      --access-point-id ${MNT_TARGET_RESOURCE_ID}   \
+      | jq -r ".MountTargets[] | select( .AvailabilityZoneName == \"${AVAILABILITY_ZONE_NAME}\" ) | .IpAddress" \
+    )
+  else
+    echo "Unsupported mount target resource: ${MNT_TARGET_RESOURCE_ID}"
+    return 1
+  fi
+
+  if [[ -z "${MOUNT_POINT_IP}" ]]
+  then
+    echo "Could not find mount target matching the current availability zone (${AVAILABILITY_ZONE_NAME}) using EFS API"
+    return 1
+  fi
+
+  DNS_NAME="${EFS_FS_ID}.efs.${AWS_REGION}.amazonaws.com"
+
+  # Backup the old hosts file
+  cp /etc/hosts "/etc/hosts.rfdk-backup-$(date +%Y-%m-%dT%H:%M:%S)"
+  # Remove any existing entries for the target DNS name
+  sed -i -e "/${DNS_NAME}/d" /etc/hosts
+  # Write the resolved entry for the target DNS name
+  cat >> /etc/hosts <<EOF
+
+${MOUNT_POINT_IP} ${DNS_NAME} # Added by RFDK
+EOF
+}
+
+# Optionally resolve DNS using the EFS API
+if [[ $RESOLVE_MOUNTPOINT_IP_VIA_API == "true"  ]]
+then
+  # jq is used to query the JSON API response
+  if ! where jq > /dev/null 2>&1
+  then
+    "${PACKAGE_MANAGER}" install -y jq
+  fi
+
+  # Get access point ID if available, otherwise file system ID
+  MNT_TARGET_RESOURCE_ID=$FILESYSTEM_ID
+  ACCESS_POINT_MOUNT_OPT=$(echo "${MOUNT_OPTIONS}" | sed -e 's#,#\n#g' | grep 'accesspoint=') || true
```

Comment:
We're creating a dependency on the format of the mount options as we define them in the MountableEfs construct here and I could see that format getting changed without testing this specific use case. Maybe we should just add a comment where we define those mount options mentioning this?

Reply:
This is parsing a standard options string as passed to the mount system call (see `-o` option in the [docs for the CLI command equivalent](https://man7.org/linux/man-pages/man8/mount.8.html)).

Discussion offline, the confusion was around the `accesspoint=...` format, which is what the [EFS mount helper expects](https://docs.aws.amazon.com/efs/latest/ug/efs-mount-helper.html#mounting-access-points) (see docs). Added a comment here with a link to those docs.